### PR TITLE
wal: releastTo should work with large release index

### PR DESCRIPTION
**Pull request by xiang90**
/cc @yichengq

### CODE REVIEW

File: wal/wal.go [hunk 1]
```diff
@@ -348,6 +348,12 @@ func (w *WAL) ReleaseLockTo(index uint64) error {
 		}
 	}
 
+	// if no lock index is greater than the release index, we can
+	// release lock upto the last one(excluding).
+	if smaller == 0 && len(w.locks) != 0 {
```
Comment (yichengq) [inline wal/wal.go]: 1. holding lock 1,2,3,4,5
2. release lock to 2
3. smaller is set to 0
4. smaller is set to len(w.locks) - 1

**Comment by yichengq** [comment]
good catch

**Comment by xiang90** [comment]
@yichengq fixed.

File: wal/wal_test.go [hunk 2]
```diff
@@ -504,4 +504,21 @@ func TestReleaseLockTo(t *testing.T) {
 			t.Errorf("#%d: lockindex = %d, want %d", i, lockIndex, uint64(i+4))
 		}
 	}
+
+	// release the lock to 5
```
Comment (yichengq) [inline wal/wal_test.go]: s/5/15

**Comment by yichengq** [comment]
lgtm

**Pull request merged**